### ISSUE TITLE
Typescript + FB 3.0 + cryptography

### ISSUE THREAD

**Issue by IagooCesaar** [hgourvest/node-firebird#234]
Hi!

I have some questions about node-firebird.

I'm using typescript on a "test projetect" and I try to connect in a **FB 3.0** database.
Here is my code to make the connection:

> import Firebird from 'node-firebird';
> import DBConnParameters from './DBConfig'
> 
> const CurrentDate = function () {
>   return new Promise(function (resolve, reject){
>     console.log('Database parameters', DBConnParameters)
>     Firebird.attachOrCreate(DBConnParameters, function (err, db) {
>       console.log('DatabaseCallback')
>       if (err) throw(err)
>       db.query('select current_date from rdb$database',[], function (err, rows) {
>         resolve(rows)
>         db.detach();
>       })
>     })
>   }
> )}
> 
> const result = CurrentDate();
> console.log(result)
> result.then((value: any) => console.log(value))

I changed my "database.conf" first and then "firebird.conf" with the adjust recommend but I cannot estabilshed a connection. I get a _pending promise_.

If I change "_DBConnParameters_" to connect in a **FB 2.5** database It works very fine.

Do I forget some adjustments or configuration to make my code works on FB 3.0 ?

And about a database cryptographed, do the library compatible with this Firebird feature?
If not, there is a planing about to include this feature?

I'm dicinding to migrate a desktop application to a web application and I want to keep my firebird database, witch will be a crypto one in a close future.

Thanks for helping!

**Comment by mariuz** [comment]
please test again with new version , see the related pull request

**Comment by mariuz** [comment]
Firebird protocol encryption is not supported yet , see the related fb 3 protocol support
